src/Engine/System/RenderSystem.cpp

```diff
@@ -12,13 +12,16 @@
 const Vector<DiffuseRenderComponent *> & RenderSystem::s_diffuseComponents(Scene::getComponents<DiffuseRenderComponent>());
 UnorderedMap<std::type_index, UniquePtr<Shader>> RenderSystem::s_shaders;
 const CameraComponent * RenderSystem::s_camera = nullptr;
+SquareShader *RenderSystem::squareShader = new SquareShader("square_vert.glsl", "square_frag.glsl");
```

Comment:
On the one hand, I'd say this shader should work like all the others - created in main and what not. On the other hand, that doesn't make sense, as post processing is integral to the rendering system and should not need to be provided by the app. Maybe shaders should just be manually defined, as having them "generically" in a vector isn't really serving any purpose at the moment. I believe @jaafersheriff is working on something to this effect. Either way, don't use new, it makes me cry. UniquePtr<SquareShader> plx

Reply:
I will change it to unique_ptr